services/app/src/actions/Search.tsx

```diff
@@ -74,22 +73,29 @@ function getSearchResults(params: SearchSettings, callback: (quests: QuestDetail
     xhr.open('POST', AUTH_SETTINGS.URL_BASE + '/quests', true);
     xhr.setRequestHeader('Content-Type', 'text/plain');
     xhr.onload = () => {
-      const response: any = JSON.parse(xhr.responseText);
-      if (response.error) {
-        dispatch({type: 'SEARCH_ERROR'});
-        return dispatch(openSnackbar(Error('Search error: ' + response.error)));
+      let response: QuestSearchResponse;
+      try {
+        response = JSON.parse(xhr.responseText);
+        if (response.error) {
```

Comment:
If catch sets response.error, should we check for response.error after the try / catch?